### ISSUE TITLE
Fix XML documentation warnings

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/msbuild#2927]
When generating XML documentation, a bunch of warnings are produced, ie CS1570, CS1572, CS1573, and CS1587.

I am disabling these warnings but ideally we should fix the underlying issues with the XML docs.

**Comment by rainersigwald** [comment]
This should be pretty easy to make progress on: remove these lines

https://github.com/microsoft/msbuild/blob/96403c598a166e5a06933f78ace75528a1d1f3b0/src/Directory.Build.props#L59-L60

then build, and fix any warnings that come up.

**Comment by vijaya-lakshmi-venkatraman** [comment]
Hi,
I see that there is a PR merged for this issue already.
If there is anything pending on this issue, I would gladly help to close it!

**Comment by Forgind** [comment]
Hi Vijaya,

Thanks for the offer! If I remember correctly, mfkl resolved all the warnings in non-deprecated code. If you're interested in fixing the rest of the issues, I'd start by removing the lines [here](https://github.com/dotnet/msbuild/blob/4f72ae4cfb09d03dccd7fc628d619ef4ab5aa479/src/Deprecated/Conversion/Microsoft.Build.Conversion.csproj#L10) and [here](https://github.com/dotnet/msbuild/blob/4f72ae4cfb09d03dccd7fc628d619ef4ab5aa479/src/Deprecated/Engine/Microsoft.Build.Engine.csproj#L16) and building. That should let you know what still needs fixing for this issue.

**Comment by Forgind** [comment]
> Hi @Forgind, Should I close the related pull request #7350? it is still open. Thanks

Yes, that would probably be best. I'm sorry I led you astray on this issue—I'll try to do better next time. Did @BenVillalobos send you the issues we discussed in triage?

**Comment by jrdodds** [comment]
Thanks @Forgind. I will also try to do better next time by communicating more.

I haven't seen a list of issues from @BenVillalobos and I would like to pick up a new issue to work on. Thanks

**Comment by BenVillalobos** [comment]
@jrdodds apologies, @Forgind is talking about https://github.com/dotnet/msbuild/issues/6752. The "higher priority up for grabs" issues will be marked as `up-for-grabs` and `grab-next`, a preexisting label we're making use of again.

For now, there's only one issue with both labels but we'll add more as we go.
